src/security.py

```
@@ -1,28 +1,232 @@
+        )
+
+    return auth_dependency
+
 
 async def auth(
     credentials: Annotated[HTTPAuthorizationCredentials, Depends(security)],
-):
-    if not USE_AUTH_SERVICE:
-        return True
-    if not credentials or credentials.credentials != SECRET_KEY:
+    db: AsyncSession = Depends(get_db),
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)